### PR TITLE
Add ability to use other API instances

### PR DESCRIPTION
**Pull request by GoodClover**
This adds the ability to use other API instances by removing the `live: bool = False` parameter from the OsmApi class and instead replacing it with `instance: str = "dev"`.
The two values `"dev"` and `"main"` point to <https://master.apis.dev.openstreetmap.org> and <https://api.openstreetmap.org> respectively, and any other value will be used as the URL.

I have not added any tests or documentation on this (I'm unsure on how you're doing the docstrings)

### CODE REVIEW

**Comment by GoodClover** [comment]
Oh sorry I just realised I forgot to make this on a different branch.
I'm not sure how to fix that, but I don't think it matters too much for this?

**Comment by jonycoo** [comment]
thanks for this addition I'll test it and add dome docu.

note: docu is in [reStructuredText](https://docutils.sourceforge.io/rst.html)

**Comment by jonycoo** [comment]
sorry for that commit mess.. 
but now it works.
thank you for your contribution.

File: pyosmapi/osm_api.py [hunk 1]
```diff
@@ -10,18 +10,22 @@
...
         if instance.lower() == "main":
             self.base_url = DEFAULT_OSM_URL
+            logger.info('Using osm main api')
         elif instance.lower() == "dev":
             self.base_url = DEFAULT_OSM_DEV_URL
+            logger.info('Using osm dev api')
         else:
-            self.base_url = instance
+            logger.warning('Key not found, using osm dev api..')
+            self.base_url = DEFAULT_OSM_DEV_URL
```
Comment (GoodClover) [inline pyosmapi/osm_api.py]: Why did you change this?
The point of my pull request is that if you set a URL instead of `dev` or `main` that is used as the API URL, meaning you can use other API instances.
